### ISSUE TITLE
YAML::TypedBadConversation error in Tutorial Sample

### ISSUE THREAD

**Issue by eminfedar** [jbeder/yaml-cpp#503]
I copied directly this sample:

```
YAML::Node lineup = YAML::Load("{1B: Prince Fielder, 2B: Rickie Weeks, LF: Ryan Braun}");
for(YAML::const_iterator it=lineup.begin();it!=lineup.end();++it) {
  //std::cout << "Playing at " << it->first.as<std::string>() << " is " << it->second.as<std::string>() << "\n";

  std::string test = it->first.as<std::string>();
}
```

and it gives that error:

```
In function `YAML::TypedBadConversion
<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::~TypedBadConversion()':
undefined reference to `vtable for YAML::BadConversion'
```

**Comment by eminfedar** [comment]
I've tried this, it gives same TypedBadConversation error too:
```
YAML::Node primes = YAML::Load("[2, 3, 5, 7, 11]");
    for (YAML::const_iterator it=primes.begin();it!=primes.end();++it) {
      qDebug() << QString::number(it->as<int>());
    }
```
Output-->
```
In function `YAML::BadConversion::BadConversion(YAML::Mark const&)':
undefined reference to `vtable for YAML::BadConversion'
```

**Comment by jbeder** [comment]
Are you sure you're linking in yaml-cpp? The right version?

**Comment by eminfedar** [comment]
I don't know what is wrong about it, I cloned latest (0.5.3) version from github and compiled it and installed it. I added "LIB += -lyaml-cpp" inside the QT .pro file. (Ubuntu 16.04, std=c++11)

**Comment by eminfedar** [comment]
This solved when I used 0.5.3 version.

**Comment by liuhang0727** [comment]
> This solved when I used 0.5.1 version.

Work!
Thanks!

**Comment by jiajihao** [comment]
I have the same problem for 0.5.3，have already been solved by:
`std::string test = static_cast<std::string>*(it->first.as<std::string>());`
